src/app/core/services/phase.service.ts

```diff
@@ -76,9 +76,22 @@ export class PhaseService {
    * If on Issue Dashboard, add previously visited repositories to otherRepos.
    * @param repo New current repository
    */
-  changeCurrentRepository(repo: Repo): void {
+  async changeCurrentRepository(repo: Repo) {
+    if (!repo) {
+      throw new Error('Invalid repo name. Please provide repo name in the format Org/Repo.');
+    }
```

Comment:
maybe this error for a invalid repo string should be thrown in `Repo::of`?

see line 19 of `repo.model.ts`

Reply:
if this is thrown in `Repo::of`, I think it's highly likely that there will be a need for wide-scale refactoring through out the app (which I think deserves an issue for itself). Let's do work on this as a separate issue